third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2089,7 +2089,8 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                 Pattern::Struct(sid, std, variant, patterns) => {
                     let mut new_inst = vec![];
                     for ty in &std.inst {
-                        let nty = subs.specialize(ty);
+                        // use `specialize_with_defaults` to get type info from constraints
+                        let nty: Type = subs.specialize_with_defaults(ty);
```

Comment:
I'm a bit concerned about using this here, as `_with_defaults` it will turn `is_integer` into `u64`, though coming up with a bad test case is hard.  Please add a test case to `pack_constraint_not_satisfied2.move` where, instead of 0, we have a literal=MAX_U64+1?